lib/fastifyCsrf.d.ts

```diff
@@ -1,37 +1,32 @@
 /// <reference types="node" />
-import * as fastify from 'fastify';
-import { Options as TokenOptions } from 'csrf';
-import 'fastify-cookie';
 
-declare module 'fastify' {
-    interface FastifyRequest<HttpRequest,
-        Query = fastify.DefaultQuery,
-        Params = fastify.DefaultParams,
-        Headers = fastify.DefaultHeaders,
-        Body = any> {
-        csrfToken(): string;
-    }
-}
+import * as http from "http";
+import * as fastify from "fastify";
+import { Options as TokenOptions } from "csrf";
+import "fastify-cookie";
 
-declare function FastifyCsrfPlugin<
-    HttpServer = any,
-    HttpRequest = any,
-    HttpResponse = any>(opts?: FastifyCsrfPlugin.Options<HttpRequest>): FastifyCsrfPlugin.Plugin<
-    HttpServer,
-    HttpRequest,
-    HttpResponse>;
+type HttpServer = http.Server;
+type HttpRequest = http.IncomingMessage;
+type HttpResponse = http.ServerResponse;
 
-declare namespace FastifyCsrfPlugin {
-    interface Options<HttpRequest> extends TokenOptions {
-        value?: (req: HttpRequest) => string;
-        cookie?: fastify.CookieSerializeOptions | boolean;
-        ignoreMethods?: string[];
-        sessionKey?: string;
-    }
+declare module "fastify" {
+  interface FastifyRequest {
+    csrfToken(): string;
+  }
+}
 
-    interface Plugin<HttpServer, HttpRequest, HttpResponse>
-        extends fastify.Plugin<HttpServer, HttpRequest, HttpResponse, FastifyCsrfPlugin.Options<HttpRequest>> {
-    }
+interface Options extends TokenOptions {
+  key?: string;
+  value?: (req: fastify.FastifyRequest) => string;
+  cookie?: fastify.CookieSerializeOptions | boolean;
+  ignoreMethods?: Array<string>;
```

Comment:
Please preserve the code style of the main code. `string[]` is preferable to to `Array<string>`, and you should use tabs for indentation instead of 2 spaces (I know, indentation is not completely consistent in the code base. Let's improve that a little bit :)).

Also, please use single quotes instead of double.

Reply:
Agreed 🙂 See 2b82425.